src/data/polynomial/eval.lean

```diff
@@ -302,9 +302,15 @@ begin
   { intros p q ph qh,
     simp only [mul_add, eval_add, ph, qh], },
   { intros n b,
-    simp [mul_assoc], }
+    simp only [mul_assoc, C_mul_monomial, eval_monomial], }
 end
 
+/-- `polynomial.eval` as linear map -/
+@[simps] def leval {R : Type*} [semiring R] (r : R) : polynomial R →ₗ[R] R :=
+{ to_fun := λ f, f.eval r,
+  map_add' := λ f g, eval_add,
+  map_smul' := λ c f, by simp only [smul_eq_C_mul, eval_C_mul, smul_eq_mul] }
```

Comment:
I think it would make sense to have an `eval_smul` lemma rather than proving it inline here

Reply:
@eric-wieser  Can you formulate it in such a way that it works for scalar multiplication by `nat` and by `R`?

Reply:
Yes, you need `is_scalar_tower S R (polynomial R)` I expect

Reply:
Turns out that `eval_smul` already existed :facepalm: :grinning: